social/app/models/post.py

```diff
@@ -120,3 +124,73 @@ def keys(tuple_list):
     Accepts a tuple list, returns a list of each tuple's first values
     """
     return [x[0] for x in tuple_list]
+
+
+def get_all_public_posts():
+    public_posts = dict()
+    public_posts["user_posts"] = Post.objects \
```

Comment:
You can simplify this logic by just returning what's returned by `Post.objects`. The actual construction of a dictionary, insertion into the `user_posts` key, and then the final return of the dictionary value for `user_posts` is redundant.